src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -149,21 +152,36 @@ private function getRelationFieldsWithoutRelationType($relations, $include_neste
     }
 
     /**
-     * Changes the BelongsTo names in the input from request to allways
-     * have the foreign_key instead of the relation name.
-     * It only changes main relations not nested.
+     * Changes the BelongsTo names in the input from request to allways have the foreign_key instead of the relation name.
+     * When $belongs_to_fields are provided we will use those fields to replace relation names on the provided input.
      *
      * eg: user -> user_id
+     *
+     * @param  array  $input
+     * @param  array  $belongsToFields
+     * @return array
      */
-    private function changeBelongsToNamesFromRelationshipToForeignKey($input)
+    private function changeBelongsToNamesFromRelationshipToForeignKey($input, $belongs_to_fields = [])
```

Comment:
This function was simple. Now it's complicated. Let's look for a way to make it simple again.